src/flow-timeout.c

```diff
@@ -246,8 +246,7 @@ static inline Packet *FlowForceReassemblyPseudoPacketSetup(Packet *p,
                                               (uint16_t *)p->tcph, 20, 0);
     }
 
-    memset(&p->ts, 0, sizeof(struct timeval));
-    TimeGet(&p->ts);
+    p->ts = f->lastts;
```

Comment:
The flows `lastts` is set to the timestamp of the last packet of the flow. If the `flow-timeout` value is for example 10 minutes, the pseudo packet could use a timestamp 10 minutes in the past. If it is larger, e.g. several hours, this gets worse.

Reply:
Wrt the pcap file use case, this will need a better explanation and analysis.

Reply:
I've updated https://redmine.openinfosecfoundation.org/issues/5037 with a pcap and some jq commands explaining the issue.

Reply:
With regards to live mode, the concept is to take the timestamp of the last packet of the flow as the actual timestamp of the transaction is just before this point in time. From a conceptual point of view, Suricata timeout the flow at lastts + N secs and Suricata creates a pseudo packet to inspect the content that is in need for processing. This packet is fake but it is an implementation "limitation"/constraint that Suricata did inspect it so late. The actual application layer traffic was really closer than lastts than it is the timeout date. So it makes more sense to use lastts instead of time().